Feature/phuongnb/comment

NguyenBat-Phuong opened the pull request.

NguyenBat-Phuong committed:
comments.ts

NguyenBat-Phuong committed:
comment

NguyenBat-Phuong committed:
include

NguyenBat-Phuong committed:
xong tag

da test thu comment, ok

khong co gi

trungntd-1508 commented on src/configs/routes/Comments.ts:
```diff
@@ -0,0 +1,11 @@
+
+const router = Router();
+
+router.post('/create/:postId/:parentId?', CommentController.create);
+router.get('/get/:postId', CommentController.get);
+router.put('/update/:id', CommentController.update);
+router.delete('/delete/:id', CommentController.delete);
```
sai routing convention

NguyenBat-Phuong committed:
like

trungntd-1508 commented on src/controllers/api/Comments.ts:
```diff
@@ -0,0 +1,107 @@
+    try {
+      const userId = req.currentUser?.userId ?? 1;
+      const { postId, parentId } = req.params;
+      const { content, taggedUserIds = [], } = req.fields || req.body;
```
sao taggedUser lại ở trong body? taggedUser cũng tương tự với hashtag là nằm trong content của comment và phải bóc tách ra từ content chứ

NguyenBat-Phuong committed:
fix comment

NguyenBat-Phuong committed:
get toggleLike

NguyenBat-Phuong committed:
findAndCountAll likes

fix tag

xoa log

comment

include

xong tag

da test thu comment, ok

khong co gi

like

fix comment

bannedHashtag

loi tao bang trung gian

21vs02

them bannedWords

xong bannedWords

xong bannedWords

validation

rvcomment

trungntd-1508 commented on src/configs/routes/Likes.ts:
```diff
@@ -0,0 +1,9 @@
+import { Router } from 'express';
```
đổi thành Reactions

trungntd-1508 commented on src/controllers/api/Comments.ts:
```diff
@@ -0,0 +1,108 @@
+  public async update(req: Request, res: Response) {
+    try {
+      const { id } = req.params;
+      if (isNaN(Number(id))) {
+        return sendError(res, 400, 'ID bình luận không hợp lệ');
+      }
```
Ko cần, cái này sẽ xử lý ở đoạn đin

trungntd-1508 commented on src/controllers/api/Comments.ts:
```diff
@@ -0,0 +1,108 @@
+      console.log(content, "content", id, "id");
+      const comment = await CommentModel.findByPk(Number(id));
+      if (!comment) {
+        return sendError(res, 404, 'Không tìm thấy bình luận');
```
sử dụng error code

trungntd-1508 commented on src/controllers/api/Comments.ts:
```diff
@@ -0,0 +1,108 @@
+        return sendError(res, 400, 'ID bình luận không hợp lệ');
+      }
+
+      const { content } = req.fields || req.body;
+      if (!content) {
+        return sendError(res, 400, 'Nội dung bình luận không được để trống');
+      }
```
ko cần, xử lý trong validation

trungntd-1508 commented on src/models/comments.ts:
```diff
@@ -0,0 +1,84 @@
+      }
+    },
+    async afterValidate(comment, options) {
+      const hashtags = comment.content.match(/#(\w+)/g)?.map(tag => tag.substring(1)) || [];
+      if (hashtags.length > 0) {
+        const allHashtags = await Promise.all(
+          hashtags.map(tag => HashtagModel.findOrCreate({ where: { name: tag } }))
+        );
+        comment.setDataValue('hashtags', allHashtags.map(([hashtag]) => hashtag));
+      }
+
+      const taggedUserIds = comment.content.match(/@(\d+)/g)?.map(mention =>
+        mention.substring(1)).filter((value, index, self) => self.indexOf(value) === index)
+      if (taggedUserIds.length > 0) {
+        comment.setDataValue('taggedUserIds', taggedUserIds);
+      }
```
hình như chưa có đoạn replace từ
"#hashtag1" về "#${hash_tag_id}". mình cần pattern dể FE biết chỗ nào là text thường và chỗ nào là hashtag có thể click vào để search

NguyenBat-Phuong committed:
taoj bang comment_reactions

NguyenBat-Phuong committed:
updateComment

NguyenBat-Phuong committed:
destroy
